modules/apps/depot/depot-test/src/testIntegration/java/com/liferay/depot/service/test/DepotEntryDLAppServiceWhenCopyingWithDocumentTypesTest.java

```diff
@@ -0,0 +1,406 @@
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(_depotGroup.getGroupId());
+
+		serviceContext.setAttribute(
+			"fileEntryTypeId", _depotDLFileEntryType.getFileEntryTypeId());
+
+		FileEntry fileEntry1 = _dlAppService.addFileEntry(
+			RandomTestUtil.randomString(), _depotGroup.getGroupId(),
+			DLFolderConstants.DEFAULT_PARENT_FOLDER_ID, _FILE_NAME,
+			ContentTypes.TEXT_PLAIN, _FILE_NAME, StringPool.BLANK,
+			StringPool.BLANK, StringPool.BLANK, new byte[0], null, null,
+			serviceContext);
+
+		FileEntry fileEntry2 = _dlAppService.copyFileEntry(
+			fileEntry1.getFileEntryId(),
+			DLFolderConstants.DEFAULT_PARENT_FOLDER_ID, _group.getGroupId(),
+			_depotDLFileEntryType.getFileEntryTypeId(),
+			_siteConnectedGroupGroupProvider.
+				getCurrentAndAncestorSiteAndDepotGroupIds(_group.getGroupId()),
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		DLFileEntry dlFileEntry1 = (DLFileEntry)fileEntry1.getModel();
+
+		DLFileEntry dlFileEntry2 = (DLFileEntry)fileEntry2.getModel();
+
+		Assert.assertEquals(
+			dlFileEntry1.getFileEntryTypeId(),
+			dlFileEntry2.getFileEntryTypeId());
+	}
+
+	@Test
+	public void testCopyFileEntryShouldNotCopyDLFileEntryTypeFromRelatedGroupUnlessDLFileEntryTypeAvailable()
+		throws Exception {
+
+		_depotEntryGroupRelLocalService.addDepotEntryGroupRel(
+			_depotEntry.getDepotEntryId(), _group.getGroupId());
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(_depotGroup.getGroupId());
+
+		serviceContext.setAttribute(
+			"fileEntryTypeId", _depotDLFileEntryType.getFileEntryTypeId());
+
+		FileEntry fileEntry1 = _dlAppService.addFileEntry(
+			RandomTestUtil.randomString(), _depotGroup.getGroupId(),
+			DLFolderConstants.DEFAULT_PARENT_FOLDER_ID, _FILE_NAME,
+			ContentTypes.TEXT_PLAIN, _FILE_NAME, StringPool.BLANK,
+			StringPool.BLANK, StringPool.BLANK, new byte[0], null, null,
+			serviceContext);
+
+		DLFileEntry dlFileEntry1 = (DLFileEntry)fileEntry1.getModel();
+
+		Assert.assertEquals(
+			_depotDLFileEntryType.getFileEntryTypeId(),
+			dlFileEntry1.getFileEntryTypeId());
+
+		Assert.assertNotEquals(
+			DLFileEntryTypeConstants.COMPANY_ID_BASIC_DOCUMENT,
+			dlFileEntry1.getFileEntryTypeId());
+
+		FileEntry fileEntry2 = _dlAppService.copyFileEntry(
+			fileEntry1.getFileEntryId(),
+			DLFolderConstants.DEFAULT_PARENT_FOLDER_ID, _group.getGroupId(),
+			DLFileEntryTypeConstants.FILE_ENTRY_TYPE_ID_BASIC_DOCUMENT,
+			_siteConnectedGroupGroupProvider.
+				getCurrentAndAncestorSiteAndDepotGroupIds(_group.getGroupId()),
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		DLFileEntry dlFileEntry2 = (DLFileEntry)fileEntry2.getModel();
+
+		Assert.assertEquals(
+			DLFileEntryTypeConstants.COMPANY_ID_BASIC_DOCUMENT,
+			dlFileEntry2.getFileEntryTypeId());
+	}
+
+	@Test
+	public void testCopyFileEntryShouldNotCopyDLFileEntryTypeFromUnrelatedGroup()
```

Comment:
Please rename it `testCopyFileEntryDoesNotCopyDLFileEntryTypeFromUnrelatedGroup`